buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -92,12 +97,26 @@ public void apply(Project project) {
             params.setDefaultParallel(findDefaultParallel(project));
             params.setInFipsJvm(Util.getBooleanProperty("tests.fips.enabled", false));
             params.setIsSnapshotBuild(Util.getBooleanProperty("build.snapshot", true));
+            params.setBwcVersions(resolveBwcVersions(rootDir));
         });
 
         // Print global build info header just before task execution
         project.getGradle().getTaskGraph().whenReady(graph -> logGlobalBuildInfo());
     }
 
+    /* Introspect all versions of ES that may be tested against for backwards
+     * compatibility. It is *super* important that this logic is the same as the
+     * logic in VersionUtils.java. */
+    private static BwcVersions resolveBwcVersions(File root) {
+        File versionsFile = new File(root, DEFAULT_VERSION_JAVA_FILE_PATH);
+        try {
+            List<String> versionLines = IOUtils.readLines(new FileInputStream(versionsFile), "UTF-8");
+            return new BwcVersions(versionLines);
+        } catch (IOException e) {
+            return null;
```

Comment:
I think we should rethrow as an `UncheckedIOException`. If we aren't able to determine BWC versions this is likely to cause all sorts of build issues so we should treat this as a fatal error. I believe this might already be the case as `BuildParams` will validate we aren't setting the value to `null`, but then we end up losing the original exception cause which is not ideal.

Reply:
I thought about it but didn’t want to fail fatal here to make testing easier. I needed to touch the testkit Test builds and wanted to keep the ability to direct configure the bwcVersion in via BuildParamd there instead of introducing sth like a test only override property in the plugin itself to pass a configurable file path to resolve versions from

Reply:
I thought about it but didn’t want to fail fatal here to make testing easier. I needed to touch the testkit Test builds and wanted to keep the ability to direct configure the bwcVersion in via BuildParamd there instead of introducing sth like a test only override property in the plugin itself to pass a configurable file path to resolve versions from

Reply:
Gotcha, we should at least log the error then, so if it does barf we get a better message than just `bwcVersions cannot be null` or whatever.

Reply:
Why does not failing with an exception make this easier to test? Tests can use expectThrows to catch an exception?

Reply:
That doesn't work with testKit tests since it's not the _test_ that's throwing the exception, it's the build and those don't get propagated to the test. You just get a "your build failed" type exception returned by teskit.